app/controllers/posts_controller.rb

```diff
@@ -7,7 +7,12 @@ def index
 
   def create
     @new_post = Post.create(post_params)
-    redirect_to '/posts'
+    if @new_post.save
+      redirect_to '\posts', notice: 'Post successful'
+    else
+      flash[:alert] = 'Post was not successful'
+      render 'index'
```

Comment:
Why might this happen? If the post is blank? If this is the only reason a post may not save is because it's blank I think the message here should be more descriptive

Reply:
We think other errors would trigger it as well eg database errors (although that's unlikely). That's why we weren't more specific in the message.